### DISCUSSION TITLE
New tag does not trigger an event

### DISCUSSION THREAD

**Discussion by ivanovaleksandar** [community/community#25973] [Actions]
<p>I would like to trigger a new action after creating a new <strong>tag</strong> in a PR close &amp; merge action:</p>
<p>So, step 1: PR close &amp; merge that creates a tag and pushes it to master successfully:</p>
<pre><code class="lang-auto">on:
  pull_request:
    branches: [ master ]
    types: [closed]

jobs:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v2  
      with:
        fetch-depth: 0
  ...

    - name: Git Tag
      if: github.event.pull_request.merged == true
      run: |
        git tag ${VERSION} ${GITHUB_SHA}
        git push --tags origin master

</code></pre>
<p>Step 2 (version a): When a new tag is pushed, initiate an event</p>
<pre><code class="lang-auto">on:
  push:
    tags:
    - '*'
</code></pre>
<p>Step 2 (version b): When a new tag is created, initiate an event</p>
<pre><code class="lang-auto">on:
  create:
    tags:
</code></pre>
<p>Even though I have tried both approaches an event is <strong>not triggered</strong>. Please help.</p>
<p><em><strong>Note</strong>: When I do manually <code>git tag</code> and <code>git push --tags</code> (not using GitHub Actions), an event (mentioned in Step 2) is triggered.</em></p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/ivanovaleksandar">@ivanovaleksandar</a>,</p>
<p>Glad to see you in Github Community Forum!</p>
<p>This is due to events triggered by the  <code>GITHUB_TOKEN</code>  will not create a new workflow run. It’s mentioned in the official doc <a href="https://docs.github.com/en/actions/configuring-and-managing-workflows/authenticating-with-the-github_token#using-the-github_token-in-a-workflow" rel="nofollow noopener">here</a>.</p>
<p>‘actions/checkout’ will persist GITHUB_TOKEN for latter git operations, hence the tag is pushed with GITHUB_TOKEN. You can add “persist-crendetials: false” to avoid this.</p>
<pre><code class="lang-auto">    - uses: actions/checkout@v2  
      with:
        fetch-depth: 0
        persist-credentials: false
</code></pre>
<p>You can create a secret wih Personal Access Token and use it to push a tag.</p>
<p>I tried to use <a href="https://developer.github.com/v3/git/tags/#create-a-tag-object" rel="nofollow noopener">rest api</a> to create a tag, code sample as below, it use PAT and can trigger workflow.</p>
<pre><code class="lang-auto">     - name: push git tag
       run: |
         curl -s -X POST https://api.github.com/repos/${{ github.repository }}/git/refs \
           -H "Authorization: token ${{ secrets.PAT1 }}" \
           -d '{ "ref": "refs/tags/v3.0.4", "sha": "shavalue" }'
</code></pre>
<p>Thanks</p>

**Comment by ivanovaleksandar** [comment]
<p><a class="mention" href="/weide-zhou">@weide-zhou</a> thanks for the reply, but whenever I try to do this with the rest api call that you suggested, I am faced with this (even though I used a PAT from an admin user with all scopes enabled):</p>
<pre><code class="lang-auto">{
  "message": "Not Found",
  "documentation_url": "https://developer.github.com/v3/git/tags/#create-a-tag-object"
}
</code></pre>
<p>What am I missing here?</p>

**Comment by weide-zhou** [comment]
<p>Hi <a class="mention" href="/ivanovaleksandar">@ivanovaleksandar</a>,</p>
<p>Thanks for your reply! What’s your code? Is there any typo?</p>
<p>My workflow for your reference:<br>
push tag: <a href="https://github.com/weide-zhou/ticket13/runs/844947896?check_suite_focus=true#step:3:45" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/runs/844947896?check_suite_focus=true#step:3:45</a><br>
2nd workflow triggered: <a href="https://github.com/weide-zhou/ticket13/runs/844948714?check_suite_focus=true" rel="nofollow noopener">https://github.com/weide-zhou/ticket13/runs/844948714?check_suite_focus=true</a></p>
<p>It’s just a sample code which use PAT token, you can change it with your case.</p>
<p>Thanks</p>

**Comment by ivanovaleksandar** [comment]
<p>Thank you <a class="mention" href="/weide-zhou">@weide-zhou</a> !</p>
<p>There were permissions missing for the call.</p>
